### PR TITLE
Format ethereum provider response in browser process instead of renderer

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/28843

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Should be covered by auto tests.

### CODE REVIEW

File: browser/brave_wallet/ethereum_provider_impl_unittest.cc [hunk 1]
```diff
@@ -2752,4 +2804,75 @@ TEST_F(EthereumProviderImplUnitTest, RequestEthCoinbase) {
...
+  const std::string error_rpc_response =
+      R"({"jsonrpc":"2.0","id":1,"error":{"code":-32005,"message":"err"}})";
+
+  // Test provider responses for send, sendAsync are in JsonRpcResponse format.
+  // And responses for request are not in JsonRpcResponse format.
+  // Success case:
+  SetInterceptor(success_rpc_response);
+  base::Value::Dict expected_dict =
+      base::test::ParseJsonDict(success_rpc_response);
+  expected_dict.Set("id", "1");
```
Comment (darkdh) [inline browser/brave_wallet/ethereum_provider_impl_unittest.cc:2821]: Please add comment saying why we need to set id as string.
https://docs.metamask.io/guide/ethereum-provider.html#legacy-methods

**Review by darkdh — APPROVED** [review]

File: browser/brave_wallet/ethereum_provider_impl_unittest.cc (hunk 1 above)
Comment (yrliou) [reply browser/brave_wallet/ethereum_provider_impl_unittest.cc:2823]: Added https://github.com/brave/brave-core/compare/2d975e6693718a1d25dc9382f18d0132f740d2ff..3ae86d20e60dd78c27359f97be8aa3d61be3b63e

**Review by StephenHeaps — APPROVED** [review]
Thank you!
Testing locally for iOS and both our issues are resolved  🎉 Created [#7244](https://github.com/brave/brave-ios/issues/7244) to update to use `SendAsync` on iOS.

[POAP joining raffle issue brave-ios #5727](https://github.com/brave/brave-ios/issues/5727):
https://user-images.githubusercontent.com/5314553/231847580-e1c58685-ef47-4878-a762-5b40bda84ef5.MP4

[Rarible connect issue brave-ios #6349](https://github.com/brave/brave-ios/issues/6349):
https://user-images.githubusercontent.com/5314553/231847646-1e23d6bb-446a-492a-9c19-92c3c177479d.MP4

**Review by kjozwiak — APPROVED** [review]
LGTM 👍 Merging as the test failures on `Win x86` are known as per https://github.com/brave/brave-core/pull/18037#issuecomment-1509097649 and not related to this specific PR.

**Pull request merged**